rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/BookKeeper.java

```diff
@@ -117,7 +117,7 @@ public BookKeeper(Configuration conf, MetricRegistry metrics) throws FileNotFoun
     initializeMetrics();
     initializeCache(conf, ticker);
     cleanupOldCacheFiles(conf);
-    fetchProcessor = new RemoteFetchProcessor(conf);
+    fetchProcessor = new RemoteFetchProcessor(this, conf);
```

Comment:
Main reason to pass the bookkeeper object here is to get rid off the thrift call from FileDownloadRequestChain. FileDownloadRequestChain has been moved to bookkeeper module. Previously we were calling setAllCached through thrift layer. With this change we already have a reference of bookkeeper so we can invoke bookkeeper.setAllCached to avoid the thrift call. 
Both these are running in the the same jvm. So avoid the call through the thrift layer

Reply:
Is this change orthogonal to the issue ? I am confused because the method `processDownloadRequests` doesnt seem to require this change. Also where will this be used ? I couldnt find a user of `processDownloadRequests` at least in this PR. Is that coming in a subsequent PR ?

Reply:
yes this change (passing bookkeeper instance) is orthogonal to the issue being reported. The change needed for that is to return totalRequestedRead instead of 0.
But while writing unit test, figured this issue and hence fixing this.